expire-old-content.js

```diff
@@ -0,0 +1,119 @@
+const fs = require('fs');
+const { execSync } = require('child_process');
+const path = require('path');
+
+// File paths
+const componentPath = './src/pages/Domov/Domov.js'; // React component with images
+const workflowPath = './.github/workflows/buildndeploy.yml'; // GitHub Actions workflow
+
+// Get today's date for comparison
+const today = new Date();
+
+// Step 1: Parse and update the React component
+function updateComponentFile() {
+    const content = fs.readFileSync(componentPath, 'utf8');
+    let hasChanges = false;
+    let removedImages = new Set();
+
+    // First pass: collect image names that should be removed
+    content.replace(/<img [^>]*src=\{([^}]+)\}[^>]*valid-until="([^"]+)"[^>]*>/g, (match, imgVar, dateStr) => {
+        const expirationDate = new Date(dateStr);
+        if (expirationDate < today) {
+            removedImages.add(imgVar.trim());
+        }
+    });
+
+    // Second pass: remove expired image elements and their trailing whitespace
+    let updatedContent = content.replace(/<img [^>]*valid-until="([^"]+)"[^>]*>(\s*(?=<|$))?/g, (match, dateStr) => {
+        const expirationDate = new Date(dateStr);
+        if (expirationDate < today) {
```

Comment:
```suggestion
        if (isNaN(expirationDate.getTime())) {
            console.log(`WARNING! Invalid date format - ${dateStr} - This will be ignored, but should be fixed ASAP.`);
            return match; // Keep the image or handle accordingly
        }
        if (expirationDate < today) {
```